modules/picard_qc/1.0/picard_qc.smk

```diff
@@ -0,0 +1,326 @@
+    resources:
+        mem_mb = CFG["mem_mb"]["alignment_summary"]
+    shell:
+        op.as_one_line("""
+        picard -Xmx{resources.mem_mb}m CollectAlignmentSummaryMetrics
+        {params.opts} 
+        I={input.bam} O={output.metrics} R={input.fasta} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+
+rule _picard_qc_insert_size:
+    input:
+        bam = rules._picard_qc_input_bam.output.sample_bam,
+    output:
+        metrics = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/insert_size_metrics",
+        histogram = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/insert_size_histogram.pdf"
+    log:
+        stderr = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/insert_size.stderr.log",
+        stdout = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/insert_size.stdout.log"
+    params:
+        opts = CFG["options"]["insert_size"]
+    conda:
+        CFG["conda_envs"]["picard"]
+    threads:
+        CFG["threads"]["insert_size"]
+    resources:
+        mem_mb = CFG["mem_mb"]["insert_size"]
+    shell:
+        op.as_one_line("""
+        picard -Xmx{resources.mem_mb}m CollectInsertSizeMetrics
+        {params.opts} 
+        I={input.bam} O={output.metrics} H={output.histogram}
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+# capture metrics
+rule _picard_qc_hs_metrics:
+    input:
+        bam = rules._picard_qc_input_bam.output.sample_bam,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        intervals = CFG["inputs"]["intervals"]
+    output:
+        hs = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics",
+        intervals = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/interval_hs_metrics"
+    log:
+        stdout = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics.stdout.log",
+        stderr = CFG["logs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/hs_metrics.stderr.log"
+    params:
+        opts = CFG["options"]["hs_metrics"]
+    conda:
+        CFG["conda_envs"]["picard"]
+    threads:
+        CFG["threads"]["hs_metrics"]
+    resources:
+        mem_mb = CFG["mem_mb"]["hs_metrics"]
+    shell:
+        op.as_one_line("""
+        picard -Xmx{resources.mem_mb}m CollectHsMetrics {params.opts} 
+        I={input.bam} O={output.hs} PER_TARGET_COVERAGE={output.intervals} 
+        R={input.fasta} TI={input.intervals} BI={input.intervals} 
+        > {log.stdout} 2> {log.stderr}
+        """)
+
+# mRNA metrics
+rule _picard_qc_rrna_int:
+    input:
+        bam = rules._picard_qc_input_bam.output.sample_bam,
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa"),
+        rrna_int = reference_files("genomes/{genome_build}/rrna_intervals/rRNA_int_gencode-33.txt")
+    output:
+        rrna_int = CFG["dirs"]["metrics"] + "{seq_type}--{genome_build}/{sample_id}/rrna_int_list"
```

Comment:
Does this generate an rRNA intervals file for every individual sample? Don't we just need to generate this once? Or is this a different file?

Reply:
oh I just realize the variable naming is misleading. The `input.rrna_int` is a reference generated only once while the `output.rrna_int` is the rrna intervals found in the sample bams. So theoutput has to be generated for each sample to run picard RNAseq metrics